ifupdown2/addons/bond.py

```diff
@@ -493,11 +511,19 @@ def get_ifla_bond_attr_from_user_config(self, ifaceobj, link_exists):
         ifla_info_data  = OrderedDict()
         old_config      = self.get_saved_ifaceobj(link_exists, ifname)
 
+        bond_mode = None
+        miimon = False
+        arp_interval = False
+
         # for each bond attribute we fetch the user configuration
         # if no configuration is provided we look for a config in policy files
         for attr_name, netlink_attr, func_ptr in self._bond_attr_set_list:
             cached_value        = None
-            user_config         = ifaceobj.get_attr_value_first(attr_name)
+            
+            if attr_name in ['bond-arp-ip-target']:
```

Comment:
This loop handles all bond generic attributes. If you want to do attribute specific processing you have to somehow do it outside. Feel free to update the data-structures to keep this as generic as possible.

Reply:
Is it ok if I copy _bond_attr_set_list and remove the bond-arp-ip-target so I can process it outside of the loop?

Reply:
Yes you can. Also please put your code in a function with an appropriate name after the generic loop

`ifaceobj.get_attr_value(attr_name)` is used when the attribute value can be provided on different line (i might be wrong but i dont think it's needed in your case).

miimon, arp_interval and bond mode can be checked outside the loop as well. If the link already exists those information should be available in the cache.

Reply:
@julienfortin how about now?
I did a hard reset of the commit to bring in the new version.